HT6_3.py

```diff
@@ -0,0 +1,240 @@
+            print(publication.expiration_date)
+        elif publication.publication_type == "Joke":
+            publication.funny_meter = publication.generate_random_funny_meter()
+            print(publication.funny_meter)
+
+
+class ReaderTxt(Reader):
+    def __init__(self):
+        super().__init__(publication_method="txt")
+        self.file_name = ""
+        self.input_list = []
+
+    def __is_valid_file_name(self, fname):
+        txt_file_list = [file for file in os.listdir() if file.endswith(f".{self.publication_method}")]
+        if fname in txt_file_list:
+            if os.path.getsize(fname) > 0:
+                return True
+            else:
+                print("File format is empty.\nPlease check folder templates_for_adding_publication for correct format.")
+        else:
+            print("File with this name is not exist. Try again.")
+
+    def get_file_name(self):
+        while True:
+            fname = f'{input("Please give name of txt file: ")}.{self.publication_method}'
+            if self.__is_valid_file_name(fname):
+                print(f"You selected file is {fname}")
+                return fname
+
+    def read_from_txt(self):
+        with open(f"{self.file_name}") as file:
+            input_list = file.readlines()
+            input_list = [row.replace('\n', '') for row in input_list]
+        while True:
+            if self.__is_valid_txt_format(input_list):
+                return input_list
+
+    def __is_valid_txt_format(self, input_list):
+        if "Publication_Type: " in input_list[0] and "Body: " in input_list[1]:
+            return True
+        else:
+            print("File format is invalid.\nPlease check folder templates_for_adding_publication for correct format.")
+            self.get_file_name()
+
+    def get_publication_type(self):
+        if "News" in self.input_list[0]:
+            print("Your publication type is News")
+            return News(publish_datetime=datetime.datetime.now(), publication_type="News")
+        elif "Private Ad" in self.input_list[0]:
+            print("Your publication type is Ad")
+            return Ad(publication_type="Ad")
+        elif "Joke" in self.input_list[0]:
+            print("Your publication type is Joke")
+            return Joke(publication_type="Joke")
+        else:
+            print(
+                f"Invalid file format. Correct format is:\n\tPublication_Type: \n\tBody: \n\tCity: or Expiration_Day:")
+
+    def get_publication_text(self):
+        body = self.input_list[1][self.input_list[1].index(': ') + 2:]
+        print(f"Publication text is:\n\t{body}")
+        return body
+
+    def get_publication_specific_info(self, publication: typing.Union[News, Ad, Joke]) -> None:
+        if publication.publication_type == "News":
+            publication.location = self.input_list[2][self.input_list[2].index(': ') + 2:]
+            print(f"News location is {publication.location}")
+            return publication.location

```

Comment:
TO REVIEW: 
I think it is redundant. Moreover the function should return None, not str